### DISCUSSION TITLE
Replace data when exporting to csv or xlsx

### DISCUSSION THREAD

**Discussion by OrakFR** [Power-Components/livewire-powergrid#1519] [Q&A] [answered]
Hello community,

I'm currently working with formatted data in specific columns of my table.

### Example:
In a customer table, I have a "contact" column that represents a relationship. Within the fields() method, I generate a link pointing to the "contact" route, and this setup works correctly for display purposes.

### Problem:
When exporting the data from the customer table, the "contact" column includes the HTML anchor tag code.
![image](https://github.com/Power-Components/livewire-powergrid/assets/129159551/0a71bf7f-0300-4d98-86c6-b7a9f73de97a)

I want the export to contain only the contact name, not the HTML.
![image](https://github.com/Power-Components/livewire-powergrid/assets/129159551/ce40c0d3-514c-46a0-8c2a-30cda7b35f66)

### Question:
Is there a way to separate the data displayed on the interface from what is exported, so that exports contain only raw data collected from the DB?

### Additional Notes:
I am aware that using a modal instead of a link could be a solution, but that approach does not meet our needs.

Thank you for any insights or suggestions you can provide!

**Comment by dansysanalyst** [comment, marked as answer]
You can create two fields, one containing the raw data and the other containing the formatted HTML link.  Let's call them `raw_contact_data` and `html_contact_data`.

Then, you then [hide](https://livewire-powergrid.com/table/include-columns.html#hidden) the column for `raw_contact_data` from table view and make it [visible for exporting](https://livewire-powergrid.com/table/include-columns.html#visibleinexport).

In addiction, you can make the column for `html_contact_data` visible for exporting and hidden in the table view.

For example:

Fields:

```php
use PowerComponents\LivewirePowerGrid\Exportable;
use PowerComponents\LivewirePowerGrid\PowerGridFields;
use PowerComponents\LivewirePowerGrid\PowerGridComponent;

final class MyPowergridTable extends PowerGridComponent
{
    use WithExport;

   //....

    public function fields(): PowerGridFields
    {
        return PowerGrid::fields()
            ->add('id')
            ->add('raw_contact_data', function (Dish $model) {
                return $model->name;
            })
            ->add('html_contact_data', function (Dish $model) {
                return '<a href="https://www.foobar.com/?q=' . e($model->name) . '"> Click for: '. e($model->name) .'</a>';
            });
}
```

Columns:

```php
use PowerComponents\LivewirePowerGrid\Exportable;
use PowerComponents\LivewirePowerGrid\Column;
use PowerComponents\LivewirePowerGrid\PowerGridComponent;

final class MyPowergridTable extends PowerGridComponent
{
    use WithExport;

   //....

 public function columns(): array
    {
        return [
            Column::make('ID', 'id')
                ->searchable()
                ->visibleInExport(false)
                ->sortable(),

            Column::make('Contact', 'html_contact_data')
                ->visibleInExport(false)
                ->sortable(),

            Column::make('Contact', 'raw_contact_data', 'name') //added "name" db field for searching"
                ->searchable()
                ->sortable()
                ->visibleInExport(true)
                ->hidden(true),
               
           Column::make('Created At', 'created_at_formatted'),
        ];
    }
}
```
The code above will produce a table containing an HTML link:

![CleanShot 2024-04-26 at 11 38 49@2x](https://github.com/Power-Components/livewire-powergrid/assets/79267265/1dfbfc16-6c0e-4a4c-a45a-de457cb6d1c2)


and an exported file containing the only name :

![CleanShot 2024-04-26 at 11 39 16@2x](https://github.com/Power-Components/livewire-powergrid/assets/79267265/5ee0fc09-9cec-4ef2-934f-b2d1ad965d96)

**Reply by OrakFR** [reply]
That will do just fine. Thank you.

**Reply by dansysanalyst** [reply]
Glad to hear @OrakFR !

________
⭐ Please consider starring our repository. It is a great way to support us!
📢 Share PowerGrid with your dev-community, friends and colleagues: Tweet, write a blog article or record a review video. 
👥 PowerGrid is free, open-source and maintained by the authors and all contributors.

**Comment by OrakFR** [comment]
I still have a question related to Columns.

When you have
`Column::make('Contact', 'raw_contact_data', 'name')`

Can I use my model's relationship db field as the 3rd argument instead of the model's field, like:
`Column::make('Contact street', 'raw_contact_data', 'contact.address.street')`
where "address" is the relationship defined in the Contact model and "street" the db column for "addresses"?

Or do I really have to use 'joins' instead (not to "practical" when you have a lot of fields)?
